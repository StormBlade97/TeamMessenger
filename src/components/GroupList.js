import GroupItem from './GroupItem'
import React from 'react'

const GroupList = (props) => {

    const groupItems = props.groupList.map(groupItem => <GroupItem key={groupItem.id} name={groupItem.name} />);

    return (
        <ul>
            {groupItems}
        </ul>
    )
}

export default GroupList