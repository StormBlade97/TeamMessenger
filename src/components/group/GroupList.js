import React from 'react'
import GroupItem from './GroupItem'

const GroupList = (props) => {

    return (
        <ul>
            {props.groupList.map(groupItem =>
                <GroupItem
                    onClick={(groupId) => {
                        props.onGroupItemSelected(groupId)
                    }}
                    key={groupItem.id}
                    id={groupItem.id}
                    name={groupItem.name}/>
            )}
        </ul>
    )
}

export default GroupList