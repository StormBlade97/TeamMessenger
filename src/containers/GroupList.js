import React from 'react'
import GroupItem from '../components/GroupItem'

export class GroupList extends React.Component {

    constructor(props) {
        super(props)
    }

    handleGroupItemClick(groupId) {
        this.props.onGroupItemSelected(groupId)
    }

    render() {
        const groupList = this.props.groupList

        return (
            <ul>
                {groupList.map(groupItem =>
                    <GroupItem
                        onClick={(groupId) => {this.handleGroupItemClick(groupId)}}
                        key={groupItem.id}
                        id={groupItem.id}
                        name={groupItem.name}/>
                )}
            </ul>
        )
    }
}

export default GroupList