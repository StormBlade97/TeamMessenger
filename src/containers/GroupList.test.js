import React from 'react'
import {GroupList} from './GroupList'
import renderer from 'react-test-renderer'

describe('Component: GroupList', () => {

    it('renders a snapshot', () => {

        const mockGroupList = [
            {
                id: 1,
                name: 'cashiers'
            },
            {
                id: 2,
                name: 'storage'
            }
        ]

        const tree = renderer.create(<GroupList groupList={mockGroupList}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})