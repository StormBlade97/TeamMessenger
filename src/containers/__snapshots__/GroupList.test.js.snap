exports[`Component: GroupList renders a snapshot 1`] = `
<ul>
  <li>
    <a
      href="#"
      onClick={[Function]}>
      cashiers
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}>
      storage
    </a>
  </li>
</ul>
`;
